app/experimenter/nimbus-ui/src/components/AppLayoutSidebarLocked/index.tsx

```diff
@@ -0,0 +1,202 @@
+import React from "react";
+import { RouteComponentProps, useParams } from "@reach/router";
+import Container from "react-bootstrap/Container";
+import Row from "react-bootstrap/Row";
+import Col from "react-bootstrap/Col";
+import Nav from "react-bootstrap/Nav";
+import Scrollspy from "react-scrollspy";
+import { AnalysisData } from "../../lib/visualization/types";
+import { analysisAvailable } from "../../lib/visualization/utils";
+import { StatusCheck } from "../../lib/experiment";
+import { DisabledItem } from "../DisabledItem";
+import { LinkNav } from "../LinkNav";
+import {
+  getExperiment_experimentBySlug_primaryProbeSets,
+  getExperiment_experimentBySlug_secondaryProbeSets,
+} from "../../types/getExperiment";
+import { ReactComponent as ChevronLeft } from "../../images/chevron-left.svg";
+import { ReactComponent as Clipboard } from "../../images/clipboard.svg";
+import { ReactComponent as BarChart } from "./bar-chart.svg";
+import LinkExternal from "../LinkExternal";
+
+export const RESULTS_LOADING_TEXT = "Checking results availability...";
+
+const getSidebarItems = (
+  metrics: { [metric: string]: string },
+  title: string,
+) => {
+  const sidebarItems = Object.keys(metrics).map((sidebarKey) => (
+    <div key={sidebarKey} className="ml-3">
+      <a href={`#${sidebarKey}`} className="inherit-color">
+        {metrics[sidebarKey]}
+      </a>
+    </div>
+  ));
+  sidebarItems.unshift(
+    <h6 key={title} className="font-weight-bold mt-3">
+      {title}
+    </h6>,
+  );
+  return sidebarItems;
+};
+
+const probesetToMapping = (
+  probesets:
+    | (getExperiment_experimentBySlug_primaryProbeSets | null)[]
+    | (getExperiment_experimentBySlug_secondaryProbeSets | null)[],
+) => {
+  const newMap: { [key: string]: string } = {};
+  probesets.map(
+    (
+      probeset:
+        | (getExperiment_experimentBySlug_primaryProbeSets | null)
+        | (getExperiment_experimentBySlug_secondaryProbeSets | null),
+    ) => (newMap[probeset!.slug] = probeset!.name),
+  );
+  return newMap;
+};
+
+type AppLayoutSidebarLockedProps = {
+  testid?: string;
+  children: React.ReactNode;
+  status: StatusCheck;
+  analysis?: AnalysisData;
+  analysisLoadingInSidebar?: boolean;
+  analysisError: Error | undefined;
```

Comment:
This can be `analysisError?: Error` to match the others since it's the same as `Error | undefined`, I'd asked before if some of these props had to be optional because I thought maybe we'd always be passing a defined value for some of them (like `analysis: AnalysisData` instead of `analysis?`) but I don't think that's true.